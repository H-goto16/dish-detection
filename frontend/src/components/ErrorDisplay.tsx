import { ThemedText } from './ThemedText';
import { ThemedView } from './ThemedView';

type ErrorDisplayProps = {
  message?: string;
  details?: string;
  className?: string;
};

/**
 * 共通のエラー表示コンポーネント
 */
const ErrorDisplay = ({
  message = '読み込みに失敗しました',
  details,
  className = '',
}: ErrorDisplayProps) => {
  return (
    <ThemedView className={`items-center justify-center p-4 ${className}`}>
      <ThemedText className='mb-1 text-center font-medium'>
        {message}
      </ThemedText>
      {details && (
        <ThemedText className='text-center text-gray-500 text-sm'>
          {details}
        </ThemedText>
      )}
    </ThemedView>
  );
};

export default ErrorDisplay;
