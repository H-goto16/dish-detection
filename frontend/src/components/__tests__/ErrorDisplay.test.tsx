import renderer from 'react-test-renderer';
import ErrorDisplay from '../ErrorDisplay';

// Mock ThemedText and ThemedView components
jest.mock('../ThemedText', () => ({
  ThemedText: ({
    children,
    ...props
  }: { children?: React.ReactNode; [key: string]: unknown }) => {
    const { Text } = require('react-native');
    return <Text {...props}>{children}</Text>;
  },
}));

jest.mock('../ThemedView', () => ({
  ThemedView: ({
    children,
    ...props
  }: { children?: React.ReactNode; [key: string]: unknown }) => {
    const { View } = require('react-native');
    return <View {...props}>{children}</View>;
  },
}));

describe('ErrorDisplay', () => {
  it('renders correctly with default message', () => {
    const tree = renderer.create(<ErrorDisplay />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with custom message', () => {
    const tree = renderer
      .create(<ErrorDisplay message='カスタムエラーメッセージ' />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with message and details', () => {
    const tree = renderer
      .create(
        <ErrorDisplay
          message='データの取得に失敗しました'
          details='ネットワーク接続を確認してください'
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with custom className', () => {
    const tree = renderer
      .create(
        <ErrorDisplay
          message='エラーが発生しました'
          className='custom-error-class'
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders without details when not provided', () => {
    const component = renderer.create(
      <ErrorDisplay message='エラーメッセージのみ' />,
    );
    const tree = component.toJSON();

    expect(tree).toBeTruthy();
    // details がない場合は1つのTextコンポーネントのみ
    if (tree && !Array.isArray(tree)) {
      expect(tree.children).toHaveLength(1);
    }
  });

  it('renders with details when provided', () => {
    const component = renderer.create(
      <ErrorDisplay message='メインメッセージ' details='詳細メッセージ' />,
    );
    const tree = component.toJSON();

    expect(tree).toBeTruthy();
    // details がある場合は2つのTextコンポーネント
    if (tree && !Array.isArray(tree)) {
      expect(tree.children).toHaveLength(2);
    }
  });

  it('handles empty strings', () => {
    const tree = renderer
      .create(<ErrorDisplay message='' details='' className='' />)
      .toJSON();
    expect(tree).toBeTruthy();
  });

  it('handles all props as undefined', () => {
    const tree = renderer
      .create(
        <ErrorDisplay
          message={undefined}
          details={undefined}
          className={undefined}
        />,
      )
      .toJSON();
    expect(tree).toBeTruthy();
  });

  it('renders without crashing', () => {
    const component = renderer.create(<ErrorDisplay />);
    expect(component).toBeTruthy();
  });
});
