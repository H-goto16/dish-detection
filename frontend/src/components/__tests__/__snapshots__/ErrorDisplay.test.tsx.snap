// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay renders correctly with default message 1`] = `
<View
  className="items-center justify-center p-4 "
>
  <Text
    className="mb-1 text-center font-medium"
  >
    読み込みに失敗しました
  </Text>
</View>
`;

exports[`ErrorDisplay renders with custom className 1`] = `
<View
  className="items-center justify-center p-4 custom-error-class"
>
  <Text
    className="mb-1 text-center font-medium"
  >
    エラーが発生しました
  </Text>
</View>
`;

exports[`ErrorDisplay renders with custom message 1`] = `
<View
  className="items-center justify-center p-4 "
>
  <Text
    className="mb-1 text-center font-medium"
  >
    カスタムエラーメッセージ
  </Text>
</View>
`;

exports[`ErrorDisplay renders with message and details 1`] = `
<View
  className="items-center justify-center p-4 "
>
  <Text
    className="mb-1 text-center font-medium"
  >
    データの取得に失敗しました
  </Text>
  <Text
    className="text-center text-gray-500 text-sm"
  >
    ネットワーク接続を確認してください
  </Text>
</View>
`;
